packages/amplify-category-function/src/provider-utils/awscloudformation/utils/cronHelper.ts

```diff
@@ -0,0 +1,139 @@
+import inquirer from 'inquirer';
+inquirer.registerPrompt('datetime', require('inquirer-datepicker'));
+import { CronBuilder } from '../utils/cronBuilder';
+
+var Days = {
+  Sunday: '1',
+  Monday: '2',
+  Tuesday: '3',
+  Wednesday: '4',
+  Thursday: '5',
+  Friday: '6',
+  Saturday: '7',
+};
+
+export async function minuteHelper(context: any) {
+  const minuteQuestion = {
+    type: 'input',
+    name: 'minutes',
+    message: 'Enter the rate in mintues:',
+    validate: context.amplify.inputValidation({
+      operator: 'regex',
+      value: '^[1-9][0-9]*$', // change to /d after checking
+      onErrorMsg: 'Value needs to be a positive integer',
+      required: true,
+    }),
+  };
+  const minuteAnswer = await inquirer.prompt([minuteQuestion]);
+  const plural = minuteAnswer.minutes === '1' ? '' : 's';
+  let cloudwatchRule = `rate(${minuteAnswer.minutes} minute${plural})`;
+  return cloudwatchRule;
+}
+
+export async function hourHelper(context: any) {
+  const hourQuestion = {
+    type: 'input',
+    name: 'hours',
+    message: 'Enter the rate in hours:',
+    validate: context.amplify.inputValidation({
+      validation: {
+        operator: 'regex',
+        value: '^[1-9][0-9]*$',
+        onErrorMsg: 'Value needs to be a positive integer',
+      },
+      required: true,
+    }),
+  };
+  const hourAnswer = await inquirer.prompt([hourQuestion]);
+  const plural = hourAnswer.hours === '1' ? '' : 's';
+  let cloudwatchRule = `rate(${hourAnswer.hours} hour${plural})`;
+  return cloudwatchRule;
+}
+
+export async function timeHelper(exp: CronBuilder) {
+  const timeQuestion = {
+    type: 'datetime',
+    name: 'dt',
+    message: 'Select the start time (use arrow keys):',
+    format: ['hh', ':', 'mm', ' ', 'A'],
+  };
+  const timeAnswer = await inquirer.prompt([timeQuestion]);
+  exp.set(
+    'minute',
+    (timeAnswer.dt as any)
+      .getMinutes()
+      .toString()
+      .split(),
+  );
+  exp.set(
+    'hour',
+    (timeAnswer.dt as any)
+      .getHours()
+      .toString()
+      .split(),
+  );
+  return exp;
+}
+
+export async function weekHelper(exp: CronBuilder) {
+  const WeekQuestion = {
+    type: 'list',
+    name: 'week',
+    message: 'Select the day to invoke the function:',
+    choices: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+  };
+  const weekAnswer = await inquirer.prompt([WeekQuestion]);
+  exp.set('dayOfTheWeek', Days[(weekAnswer as any).week].split());
+  return exp;
+}
+
+export async function monthHelper(exp, context) {
+  const dateQuestion = {
+    type: 'datetime',
+    name: 'dt',
+    message: 'Select on which day of the month to invoke the function (use arrow keys):',
+    format: ['DD'],
+  };
+  const dateAnswer = await inquirer.prompt([dateQuestion]);
```

Comment:
can't we decounstuct `dt` from the prompt result and cast it to something to save all the `any` casts all around? even if it is `any` would be readable with a single cast.

Reply:
I use type to cast "any" and  used that type instead of using any